CONTRIBUTING.rst

```diff
@@ -0,0 +1,258 @@
+    $ git push origin name-of-your-bugfix-or-feature
+
+  If ``pre-commit`` hooks fail, try re-committing your changes (or, if need be, you can skip them with `$ git commit --no-verify`).
+
+#. Submit a `Pull Request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_ through the GitHub website.
+
+#. When pushing your changes to your branch on GitHub, the documentation will automatically be tested to reflect the changes in your Pull Request. This build process can take several minutes at times. If you are actively making changes that affect the documentation and wish to save time, you can compile and test your changes beforehand locally with::
+
+    # To generate the html and open it in your browser
+    $ make docs
+    # To only generate the html
+    $ make autodoc
+    $ make -C docs html
+    # To simply test that the docs pass build checks
+    $ tox -e docs
+
+#. Once your Pull Request has been accepted and merged to the ``main`` branch, several automated workflows will be triggered:
+
+    - The ``bump-version.yml`` workflow will automatically bump the patch version when pull requests are pushed to the ``main`` branch on GitHub. **It is not recommended to manually bump the version in your branch when merging (non-release) pull requests (this will cause the version to be bumped twice).**
+    - `ReadTheDocs` will automatically build the documentation and publish it to the `latest` branch of `xncml` documentation website.
+    - If your branch is not a fork (ie: you are a maintainer), your branch will be automatically deleted.
+
+  You will have contributed your first changes to ``xncml``!
+
+Pull Request Guidelines
+-----------------------
+
+Before you submit a pull request, check that it meets these guidelines:
+
+#. The pull request should include tests and should aim to provide `code coverage <https://en.wikipedia.org/wiki/Code_coverage>`_ for all new lines of code. You can use the ``--cov-report html --cov xncml`` flags during the call to ``pytest`` to generate an HTML report and analyse the current test coverage.
+
+#. If the pull request adds functionality, the docs should also be updated. Put your new functionality into a function with a docstring, and add the feature to the list in ``README.rst``.
+
+#. The pull request should work for Python 3.8, 3.9, 3.10, 3.11, and 3.12. Check that the tests pass for all supported Python versions.
+
+Tips
+----
+
+To run a subset of tests::
+
+$ pytest tests.test_xncml
+
+To run specific code style checks::
+
+    $ black --check xncml tests
+    $ isort --check xncml tests
+    $ blackdoc --check xncml docs
+    $ ruff xncml tests
+    $ flake8 xncml tests
+
+To get ``black``, ``isort``, ``blackdoc``, ``ruff``, and ``flake8`` (with plugins ``flake8-alphabetize`` and ``flake8-rst-docstrings``) simply install them with `pip` into your environment.
```

Comment:
Why not refer all to make, so the user doesn't have to worry about the specific details, and we don't have to keep this updated with the code. 

The user should have to do manual steps. `make dev` should install anything necessary.